src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        Shape[] shapes = new Shape[6];
+
+        for (int i = 0; i < shapes.length / 2; i++) {
+            shapes[i] = new ShapeSupplier().getRandomShape();
```

Comment:
According to the checklist, you should not create a new `ShapeSupplier` instance inside the loop. Instead, create it once before the loop starts and reuse it. This will improve performance and adhere to best practices.